### PR TITLE
[New Rule] Unusual User Calling the Metadata Service [Linux]

### PR DESCRIPTION
**Pull request by randomuserid**
## Issues

https://github.com/elastic/mechagodzilla/issues/103

## Summary

ML rule for the Linux version of the job that looks for rare user name calling the metadata service

### CODE REVIEW

File: rules/ml/ml_linux_anomalous_metadata_user.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[metadata]
+creation_date = "2020/09/22"
+ecs_version = ["1.6.0"]
+maturity = "development"
```
Comment (bm11100) [inline rules/ml/ml_linux_anomalous_metadata_user.toml:4]:
```suggestion
maturity = "production"
```

File: rules/ml/ml_linux_anomalous_metadata_user.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+updated_date = "2020/09/22"
+
+[rule]
+anomaly_threshold = 75
+author = ["Elastic"]
+description = "Looks for anomalous access to the metadata service by an unusual user. The metadata service may be targeted in order to harvest credentials or user data scripts containing secrets."
+false_positives = ["A newly installed program, or one that runs under a new or rarely used user context, could trigger this detection rule. Manual interrogation of the metadata service during debugging or troubleshooting could trigger this rule."]
```
Comment (bm11100) [inline rules/ml/ml_linux_anomalous_metadata_user.toml:11]: needs linting

**Review by bm11100 — CHANGES_REQUESTED** [review]

**Review by bm11100 — APPROVED** [review]

File: rules/ml/ml_linux_anomalous_metadata_user.toml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+creation_date = "2020/09/22"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/09/22"
+
+[rule]
+anomaly_threshold = 75
+author = ["Elastic"]
+description = """
+Looks for anomalous access to the metadata service by an unusual user. The metadata service may be targeted in order to
```
Comment (brokensound77) [inline rules/ml/ml_linux_anomalous_metadata_user.toml:11]: what is the metadata service?

Comment (brokensound77) [reply rules/ml/ml_linux_anomalous_metadata_user.toml:11]: Ah I see from the issue. Should we be more specific in the description and name that it is AWS Metadata Service

Comment (randomuserid) [reply rules/ml/ml_linux_anomalous_metadata_user.toml:11]: It actually is a feature of AWS, GCP and Azure. They all have it.

Comment (devonakerr) [reply rules/ml/ml_linux_anomalous_metadata_user.toml:11]: maybe specify "cloud platform metadata service"?

File: rules/ml/ml_linux_anomalous_metadata_user.toml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+harvest credentials or user data scripts containing secrets.
```
Comment (bm11100) [inline rules/ml/ml_linux_anomalous_metadata_user.toml:12]:
```suggestion
Looks for anomalous access to the cloud platform metadata service by an unusual user. The metadata service may be targeted in order to
harvest credentials or user data scripts containing secrets.
```

Comment (bm11100) [reply rules/ml/ml_linux_anomalous_metadata_user.toml:12]: may need to re-lint after this change

**Review by devonakerr — APPROVED** [review]

**Pull request merged**